<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Pedro Freire (www.pedrofreire.com)</author>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>430</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CG Maze (Labirinto)</string>
  </property>
  <property name="windowIcon">
   <iconset resource="imgs.qrc">
    <normaloff>:/mainwindow/imgs/mainwindow.icon.png</normaloff>:/mainwindow/imgs/mainwindow.icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLabel" name="label">
      <property name="font">
       <font>
        <pointsize>12</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="toolTip">
       <string/>
      </property>
      <property name="text">
       <string>Labirinto</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horiz">
      <item>
       <widget class="QTabWidget" name="tabSet">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="contextMenuPolicy">
         <enum>Qt::NoContextMenu</enum>
        </property>
        <property name="tabPosition">
         <enum>QTabWidget::South</enum>
        </property>
        <property name="tabShape">
         <enum>QTabWidget::Rounded</enum>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <property name="elideMode">
         <enum>Qt::ElideNone</enum>
        </property>
        <property name="usesScrollButtons">
         <bool>false</bool>
        </property>
        <widget class="QWidget" name="tabMap">
         <attribute name="title">
          <string>&amp;Mapa</string>
         </attribute>
         <attribute name="toolTip">
          <string>Vista de cima do labirinto (mapa)</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QFrame" name="frame">
            <property name="frameShape">
             <enum>QFrame::Box</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <widget class="ViewMapGL" name="glViewMap" native="true">
               <property name="contextMenuPolicy">
                <enum>Qt::NoContextMenu</enum>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <spacer name="horizontalSpacer_4">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushNew">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>26</height>
               </size>
              </property>
              <property name="contextMenuPolicy">
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip">
               <string>Começar de novo com outro labirinto</string>
              </property>
              <property name="text">
               <string>&amp;Novo</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabFP2D">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <attribute name="title">
          <string>1ª Pessoa &amp;2D</string>
         </attribute>
         <attribute name="toolTip">
          <string>Vista de dentro do labirinto (com OpenGL 2D)</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QFrame" name="frame_2">
            <property name="frameShape">
             <enum>QFrame::Box</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_6">
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <widget class="View2DGL" name="glView2D" native="true">
               <property name="contextMenuPolicy">
                <enum>Qt::NoContextMenu</enum>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushTurnLeft2D">
              <property name="minimumSize">
               <size>
                <width>32</width>
                <height>24</height>
               </size>
              </property>
              <property name="contextMenuPolicy">
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip">
               <string>Virar à esquerda</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="imgs.qrc">
                <normaloff>:/mainwindow/imgs/mainwindow.icon.turnLeft.png</normaloff>:/mainwindow/imgs/mainwindow.icon.turnLeft.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>11</width>
                <height>9</height>
               </size>
              </property>
              <property name="shortcut">
               <string notr="true">Left</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushMoveFwd2D">
              <property name="minimumSize">
               <size>
                <width>48</width>
                <height>26</height>
               </size>
              </property>
              <property name="contextMenuPolicy">
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip">
               <string>Avançar</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="imgs.qrc">
                <normaloff>:/mainwindow/imgs/mainwindow.icon.goFwd.png</normaloff>:/mainwindow/imgs/mainwindow.icon.goFwd.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>11</width>
                <height>11</height>
               </size>
              </property>
              <property name="shortcut">
               <string notr="true">Up</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushTurnRight2D">
              <property name="minimumSize">
               <size>
                <width>32</width>
                <height>24</height>
               </size>
              </property>
              <property name="contextMenuPolicy">
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip">
               <string>Virar à direita</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="imgs.qrc">
                <normaloff>:/mainwindow/imgs/mainwindow.icon.turnRight.png</normaloff>:/mainwindow/imgs/mainwindow.icon.turnRight.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>11</width>
                <height>9</height>
               </size>
              </property>
              <property name="shortcut">
               <string notr="true">Right</string>
              </property>
              <property name="flat">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabFP3D">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <attribute name="title">
          <string>1ª Pessoa &amp;3D</string>
         </attribute>
         <attribute name="toolTip">
          <string>Vista de dentro do labirinto (com OpenGL 3D)</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QFrame" name="frame_3">
            <property name="frameShape">
             <enum>QFrame::Box</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <widget class="View3DGL" name="glView3D" native="true">
               <property name="contextMenuPolicy">
                <enum>Qt::NoContextMenu</enum>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <spacer name="horizontalSpacer_9">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushTurnLeft3D">
              <property name="minimumSize">
               <size>
                <width>32</width>
                <height>24</height>
               </size>
              </property>
              <property name="contextMenuPolicy">
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip">
               <string>Virar à esquerda</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="imgs.qrc">
                <normaloff>:/mainwindow/imgs/mainwindow.icon.turnLeft.png</normaloff>:/mainwindow/imgs/mainwindow.icon.turnLeft.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>11</width>
                <height>9</height>
               </size>
              </property>
              <property name="shortcut">
               <string notr="true">Left</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushMoveFwd3D">
              <property name="minimumSize">
               <size>
                <width>48</width>
                <height>26</height>
               </size>
              </property>
              <property name="contextMenuPolicy">
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip">
               <string>Avançar</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="imgs.qrc">
                <normaloff>:/mainwindow/imgs/mainwindow.icon.goFwd.png</normaloff>:/mainwindow/imgs/mainwindow.icon.goFwd.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>11</width>
                <height>11</height>
               </size>
              </property>
              <property name="shortcut">
               <string notr="true">Up</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushTurnRight3D">
              <property name="minimumSize">
               <size>
                <width>32</width>
                <height>24</height>
               </size>
              </property>
              <property name="contextMenuPolicy">
               <enum>Qt::NoContextMenu</enum>
              </property>
              <property name="toolTip">
               <string>Virar à direita</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="imgs.qrc">
                <normaloff>:/mainwindow/imgs/mainwindow.icon.turnRight.png</normaloff>:/mainwindow/imgs/mainwindow.icon.turnRight.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>11</width>
                <height>9</height>
               </size>
              </property>
              <property name="shortcut">
               <string notr="true">Right</string>
              </property>
              <property name="flat">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_10">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabAbout">
         <attribute name="title">
          <string>&amp;Autoria</string>
         </attribute>
         <attribute name="toolTip">
          <string>Autoria do jogo</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
a, a:visited { text-decoration: none }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;img src=&quot;:/mainwindow/imgs/mainwindow.icon.png&quot; /&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:8pt;&quot;&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Jogo do labirinto para Computação Gráfica, v1.1&lt;br /&gt;Exemplo e exercício académico&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:8pt;&quot;&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:8pt;&quot;&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;© 2010 Pedro Freire&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;a href=&quot;http://www.pedrofreire.com/&quot; title=&quot;Visitar o site do autor&quot;&gt;www.pedrofreire.com&lt;/a&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:8pt;&quot;&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;a href=&quot;http://creativecommons.org/licenses/by-nc-sa/2.5/pt/&quot; title=&quot;Ler o texto completo da licença&quot;&gt;&lt;img src=&quot;:/mainwindow/imgs/mainwindow.icon.cc.png&quot; /&gt;&lt;/a&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:8pt;&quot;&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Este software é gratuito &lt;b&gt;desde que&lt;/b&gt;&lt;br&gt;mantenha esta mensagem de direito de autor.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:8pt;&quot;&gt;&lt;/p&gt;
&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="openExternalLinks">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="font">
           <font>
            <family>MS Serif</family>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="toolTip">
           <string>Norte</string>
          </property>
          <property name="text">
           <string>N</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDial" name="dialCompass">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="toolTip">
           <string>Orientação da vista</string>
          </property>
          <property name="maximum">
           <number>4</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="pageStep">
           <number>1</number>
          </property>
          <property name="value">
           <number>2</number>
          </property>
          <property name="sliderPosition">
           <number>2</number>
          </property>
          <property name="tracking">
           <bool>false</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="invertedAppearance">
           <bool>false</bool>
          </property>
          <property name="invertedControls">
           <bool>false</bool>
          </property>
          <property name="wrapping">
           <bool>true</bool>
          </property>
          <property name="notchTarget">
           <double>1.000000000000000</double>
          </property>
          <property name="notchesVisible">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="font">
           <font>
            <family>MS Serif</family>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="toolTip">
           <string>Sul</string>
          </property>
          <property name="text">
           <string>S</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>640</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ViewMapGL</class>
   <extends>QWidget</extends>
   <header>view_gl.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>View2DGL</class>
   <extends>QWidget</extends>
   <header>view_gl.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>View3DGL</class>
   <extends>QWidget</extends>
   <header>view_gl.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabSet</tabstop>
  <tabstop>pushNew</tabstop>
  <tabstop>pushTurnLeft2D</tabstop>
  <tabstop>pushMoveFwd2D</tabstop>
  <tabstop>pushTurnRight2D</tabstop>
  <tabstop>dialCompass</tabstop>
 </tabstops>
 <resources>
  <include location="imgs.qrc"/>
 </resources>
 <connections/>
</ui>
